gossipd/routing.c

```diff
@@ -1047,9 +1047,10 @@ u8 *handle_channel_update(struct routing_state *rstate, const u8 *update)
 	if (!routing_add_channel_update(rstate, serialized))
 		status_failed(STATUS_FAIL_INTERNAL_ERROR,
 			      "Failed adding channel_update");
-	/* Only store updates for public channels */
-	if (chan->public)
-		gossip_store_add_channel_update(rstate->store, serialized);
+	/* Store the channel_update for both public and non-public channels
+	 * (non-public ones may just be the incoming direction. We'd have
+	 * dropped invalid ones earlier. */
+	gossip_store_add_channel_update(rstate->store, serialized);
```

Comment:
The earlier commit, `create_channel_update` in `channeld/channel.c` creates a `channel_update` with all-0s signature.  And above this, there is a check that the signature matches (call to `check_channel_update`).  Will such a "local channel update" created by `channeld` reach this point?

Reply:
The comment on the `send_temporary_announcement` was outdated: we don't send a `channel_announcement`, instead just telling `gossipd` with a `local_add_channel`, and the `channel_update` is signed, in order to avoid having to special case this. It's also what we discussed during last week's call, and others are happy to use a fully signed update for this. It won't make much sense to nodes that haven't seen the `channel_announcement` anyway :wink: 

Changed the stale function comment for `send_temporary_announcement` to reflect these changes.